pkg/apis/kubeone/helpers.go

```diff
@@ -279,6 +279,8 @@ func (p CloudProviderSpec) CloudProviderName() string {
 		return "equinixmetal"
 	case p.Vsphere != nil:
 		return "vsphere"
+	case p.VMwareCloudDirector != nil:
+		return "vmware-cloud-director"
```

Comment:
Shouldn't this return the JSON tag (`vmwareCloudDirector`)?

Reply:
This is the expected value.

Reply:
I'm worried that this might break something because we use this function in some other places as well (and we might start using it in other places too). What if we add a new function to determine the CloudProviderName for machine-controller? Something like this:

```go
func MachineControllerCloudProvider(p CloudProviderSpec) {
    switch {
      case p.VMwareCloudDirector != nil:
        return "vmware-cloud-director"
      default:
        return p.CloudProviderName()
    }
}
```

Reply:
@kron4eg PTAL ^^

Reply:
Done